toqito/rand/tests/test_random_density_matrix.py

```diff
@@ -1,35 +1,54 @@
 """Test random_density_matrix."""
 import numpy as np
+import pytest
 
 from toqito.matrix_props import is_density
 from toqito.rand import random_density_matrix
 
 
-def test_random_density_not_real():
-    """Generate random non-real density matrix."""
-    mat = random_density_matrix(2)
-    np.testing.assert_equal(is_density(mat), True)
-
-
-def test_random_density_real():
-    """Generate random real density matrix."""
-    mat = random_density_matrix(2, True)
-    np.testing.assert_equal(is_density(mat), True)
-
-
-def test_random_density_not_real_bures():
-    """Random non-real density matrix according to Bures metric."""
-    mat = random_density_matrix(2, distance_metric="bures")
-    np.testing.assert_equal(is_density(mat), True)
-
-
-def test_random_density_not_real_k_param():
-    """Generate random non-real density matrix wih k_param."""
-    mat = random_density_matrix(2, distance_metric="bures")
-    np.testing.assert_equal(is_density(mat), True)
-
-
-def test_random_density_not_real_all_params():
-    """Generate random non-real density matrix all params."""
-    mat = random_density_matrix(2, True, 2, "haar")
-    np.testing.assert_equal(is_density(mat), True)
+@pytest.mark.parametrize("dim, is_real, k_param, distance_metric", [
+    # Generate random non-real density matrix.
+    (2, False, None, "haar"),
+    # Generate random real density matrix.
+    (2, True, None, "haar"),
+    # Random non-real density matrix according to Bures metric.
+    (2, False, None, "bures"),
+    # Generate random non-real density matrix all params.
+    (2, True, 2, "haar"),
+])
+def test_random_density(dim, is_real, k_param, distance_metric):
+    if k_param == dim:
+        mat = random_density_matrix(
+            dim=dim, is_real=is_real, k_param=k_param, distance_metric=distance_metric
+        )
+        np.testing.assert_equal(is_density(mat), True)
+
+
+def is_positive_semidefinite(matrix):
+    """Check if a matrix is positive semidefinite."""
+    return np.all(np.linalg.eigvals(matrix) >= 0)
```

Comment:
I might be mistaken but where is `matrix` defined?

Reply:
Not mistaken. Some weird copy/paste thing here on my part. Removing, nice catch!